src/sqs/SqsClientPlugin.ts

```diff
@@ -0,0 +1,27 @@
+// tslint:disable:prefer-function-over-method
+import { BaseSingletonDefinition } from '../ioc/objectdefinition/BaseSingletonDefinition';
+import BaseApp, { Plugin } from '../app/BaseApp';
+import { SqsClient } from './SqsClient';
+
+export default class SqsClientPlugin implements Plugin {
+  name = 'SqsClientPlugin';
+
+  getName() {
+    return this.name;
+  }
+
+  willStart(app: BaseApp) {
+    if (!app.hasConfig('SqsClient')) {
+      throw new Error('SqsClientPlugin has been registered but could not find config using key "mysql"');
+    }
+
+    const context = app.getContext();
+    const confs = context.getConfig('SqsClient');
+    Object.keys(confs).forEach((key) => {
+      const clientSingleton = new BaseSingletonDefinition<any>(SqsClient, key);
+      clientSingleton.constructorParamByValue(confs[key]);
```

Comment:
The constructor for SqsClient has no parameters, so in this declaration you shouldn't pass constructor params. I see that instead, you're allowing for a queueUrl, name and awsRegion. You should either put all those inside of a configuration object, or you should pass them as setPropertyByValue().
I think the config object is way better as it is extensible in the future.

Reply:
THis is fine now. Thanks 👍